specific/cmo_sale/models/sale.py

```diff
@@ -67,39 +68,66 @@ class SaleOrder(models.Model):
         states={'done': [('readonly', True)]},
         default=lambda self: self._get_th_convenant(),
     )
+    discount_type = fields.Selection(
+        [('percent', 'Percentage'),
+         ('amount', 'Amount')],
+         inverse='_compute_discount_rate',
+         string='Discount Type',
+         states={'done': [('readonly', True)]},
+    )
+    discount_rate = fields.Float(
+        string='Discount Rate',
+        inverse='_compute_discount_rate',
+        store=True,
+        states={'done': [('readonly', True)]},
+    )
+    amount_discount = fields.Float(
+        string="Discount",
+        compute='_compute_amount_discount',
+        readonly=True,
+    )
+
+    @api.depends('amount_untaxed')
```

Comment:
Is this compute field is for @api.one or @api.multi ?
If you not specify it should assume @api.multi. Which means, it need **for rec in self:**